src/api/prop.js

```diff
@@ -11,27 +12,99 @@ export function create (def) {
   };
 }
 
+
+
+// array
+
 export const array = create({
   coerce: val => (Array.isArray(val) ? val : [val]),
   default: () => [],
   deserialize: JSON.parse,
   serialize: JSON.stringify
 });
 
+
+
+// boolean
+
 export const boolean = create({
   coerce: value => !!value,
   default: false,
   deserialize: value => !(value === null),
   serialize: value => (value ? '' : undefined)
 });
 
+
+
+// number
+
 export const number = create({
   default: 0,
   coerce: alwaysUndefinedIfNotANumberOrNumber,
   deserialize: alwaysUndefinedIfNotANumberOrNumber,
   serialize: alwaysUndefinedIfNotANumberOrNumber
 });
 
+
+
+// slot
+
+const { MutationObserver } = window;
+const attrSlot = 'slot';
```

Comment:
I think I was thrown off by the names of the variables. I read the cost top to bottom and stumbled on all of them. Perhaps I didn't have enough context to the problem-being-solved before reading it.